test/dynamo/test_nested_graph_breaks.py

```diff
@@ -943,6 +943,57 @@ def f4(x):
         # multiplication by 32, 64, 128, 256
         self.assertEqual(cnts.op_count, 4)
 
+    def test_error_on_graph_break_nested(self):
+        # error_on_graph_break in a nested frame
+        cnts = torch._dynamo.testing.CompileCounter()
+
+        @torch._dynamo.error_on_graph_break(False)
+        def inner_f5(x):
+            x = x + 2
+            torch._dynamo.graph_break()
+            return x + 4
+
+        @torch._dynamo.error_on_graph_break(True)
```

Comment:
For completeness, can we have a test covering True in the outer and True in the inner?  This should be covered but want to make sure our unit tests covers it